### ISSUE TITLE
Modifiyng hard link zettlr deletes file contents

### ISSUE THREAD

**Issue by 0range0range** [Zettlr/Zettlr#4521]
### Description

When I have a file opened in Zettlr and it's hardlink opened with another editor, when I delete content in another editor, Zettlr often (like 50%/50%) deletes the content of the file completely.

What's worse, if I sit idle for some time, autosave is made which now changes inode to blank file. Ctrl-Z does not work.
If I press save in another text editor before autosave, content gets reloaded into zettlr.

When **adding** the content to a file from outside, I did not see this problem.
I sometimes want to have some notes in other folders so that changes are always propagated (1 version). Also the same is happening with soft links.

### Reproducing

1. Create file in Zettlr.
2. Create hard link to this file elsewhere
3. Open hard link in another text editor (I did with gedit)
4. Try to delete content from hard link and switch to Zettlr (not sure about the timing)
5. File content deleted (empty file)

### Zettlr Version

Stable (most recent version)

### Specify version

Zettlr-3.0.0-beta.6

### Installation Method

Other (please specify below)

### Your Platform

- [ ] Windows
- [ ] macOS
- [X] Linux

### Architecture

- [X] Intel 64bit
- [ ] ARM 64bit

### Operating System Version

Arch

### Additional Information

Other ways to achieve the functionality are also welcome.

**Comment by nathanlesage** [comment]
Could you please provide more info on how you created this hard link? I rarely use links, so I'll have to dig into what is happening when editing hard linked files externally.

**Comment by 0range0range** [comment]
```bash
ln real_file link_name 
```

```
ln -s real_file link_name 
```

**Comment by nathanlesage** [comment]
If I understand it correctly, a "hard link" is basically just an alias filename for the very same data on disk, right? It may prove difficult to account for that. I can't promise anything, but if someone else wants to give it a shot, be my guest.

**Comment by 0range0range** [comment]
> If I understand it correctly, a "hard link" is basically just an alias filename for the very same data on disk, right? It may prove difficult to account for that. I can't promise anything, but if someone else wants to give it a shot, be my guest.

Yep. The key with "hard link" is one can delete an original and the link will be perfectly functional as a separate file. But if changes are done to the hard link it is changing actual bits so original will also be changed.

**Comment by 0range0range** [comment]
@AkechiShiro Thanks for your interest. Yes, you are correct. To give more background:

My end use case is basically that I have a zettelkasten, but some of the notes I want to reuse in active projects and often I change them during project. So I wanted to have a note in both the zettelkasten, but also at hand in the active project folder.

In order not to have 2 different versions of the same file in 2 places, I thought that hard links might solve it and decided to test, having hard link and original file opened in different editors side by side.  While testing I observed this behaviour.

My main use case is not to have parallel editing of files. Rather, I am worried here that I would do so by accident and the file contents will get deleted and overwritten to the disk by autosave. Expected behaviour would be, that if I erase a word in other text editor (let's say from hard link), this modifies actual bits on the disk, which Zettlr then fetches in after some time.

**Comment by AkechiShiro** [comment]
I'll let @nathanlesage explain if there is a solution for your complex use case, on my side, I'd say Zettlr would need to listen to filesystem changes or changes made to the file, I'm not familiar with how Zettlr works right now, so I can't say if that's easy to do or not

**Comment by nathanlesage** [comment]
I have now re-tested this, and while there seem to be some instances where Zettlr doesn't pick up the changes immediately, there is no data loss anymore!
